tests/testCountInputs.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python
+#
+# LSST Data Management System
+# Copyright 2008-2015 LSST/AURA
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+"""
+Tests for CountInputs measurement algorithm
+"""
+
+import unittest
+import lsst.utils.tests as utilsTests
+
+import lsst.pex.config as pexConfig
+import lsst.afw.geom as afwGeom
+import lsst.afw.table as afwTable
+import lsst.afw.image as afwImage
+import lsst.afw.coord as afwCoord
+import lsst.afw.detection as afwDetection
+import lsst.meas.base as measBase
+from lsst.afw.geom.polygon import Polygon
+
+
+class CountInputsTest(unittest.TestCase):
+    def testCountInputs(self):
```

Comment:
There's good news and there's bad news here.

The good news: I really like that this test is a serious attempt to exercise the new functionality. Good job!

The bad news: the code is long, confusing and difficult to read.

It doesn't help that the results are a bit unexpected: you have three sources and three input images. One of your sources appears in two inputs, one in one input, and one in none of the inputs. This seems bizarre -- how can you have a source in your coadd that was in none of the inputs?! -- but I'm not sure if it's intentional.

I suggest two specific changes which I think would help:

1. You set `num = 3` at the top of the method, and comment `Number of images`. In fact, it seems like this is _both_ the number of images _and_ the number of test sources. This degeneracy is confusing: if you really want to use `num` for both these quantities, at least say so in the comments. Better still, define two different variables.
2. The code should be better commented throughout. In particular:
  * Add a block at the top, describing the overall aim of the test and summarizing what you expect the results to be. This would be a good place to mention which inputs you expect the sources to appear in, for example;
  * Before each section of the code (setting up the image, setting up the CCDs, adding the sources, running the measurement task), and a brief comment describing what's happening.